### PR TITLE
refactor(publications): fixed eslint errors

### PR DESCRIPTION
**Pull request by bodnara**
* added missing/more specific types to variables
* added missing/more specific return types
* eliminated assigment of any values
* handled unused Promise values

### CODE REVIEW

File: apps/publications/src/app/dialogs/add-authors-dialog/add-authors-dialog.component.ts [hunk 1]
```diff
@@ -45,21 +31,33 @@ export class AddAuthorsDialogComponent implements OnInit {
...
+  onSearchByString(): void {
     this.searchLoading = true;
     const removeAuthors = [...this.alreadyAddedAuthors, ...this.authorsToAdd];
-    this.cabinetService.findNewAuthors(this.searchControl.value).subscribe(
+    this.cabinetService.findNewAuthors(this.searchControl.value as string).subscribe(
       (authors) => {
-        authors = authors.filter(
-          (item) => removeAuthors.map((author) => author.id).indexOf(item.id) <= -1
-        );
+        authors = authors.filter((item) => removeAuthors.includes(item));
```
Comment (xkureck) [inline apps/publications/src/app/dialogs/add-authors-dialog/add-authors-dialog.component.ts:60]: This filter is not working
when i put to searchfield something like "martin" the backend post request finds many possible author but the table says No authors were found.
image

File: apps/publications/src/app/dialogs/add-thanks-dialog/add-thanks-dialog.component.ts [hunk 2]
```diff
@@ -27,37 +33,29 @@ export class AddThanksDialogComponent implements OnInit {
...
-  tableId = TABLE_ADD_THANKS_DIALOG;
-
   ngOnInit(): void {
     this.loading = true;
-    const allowedOwners = this.storeService.get('allowed_owners_for_thanks');
+    const allowedOwners = this.storeService.get('allowed_owners_for_thanks') as number[];
     this.ownersManagerService.getAllOwners().subscribe((owners) => {
       if (allowedOwners.length !== 0) {
-        this.owners = owners.filter((item) => allowedOwners.indexOf(item.id) > -1);
+        this.owners = owners.filter((item) => allowedOwners.includes(item.id));
```
Comment (xkureck) [inline apps/publications/src/app/dialogs/add-thanks-dialog/add-thanks-dialog.component.ts:41]: I dont think this filtering works, even if I insert something into the allowed_owners_for_thanks, there is no owner I can add
also I wanted to be the default value - when the field is empty - all owners can be thanked

File: apps/publications/src/app/dialogs/add-thanks-dialog/add-thanks-dialog.component.ts [hunk 3]
```diff
@@ -27,37 +33,29 @@ export class AddThanksDialogComponent implements OnInit {
...
       } else {
         this.owners = owners;
       }
-      this.owners = this.owners.filter(
-        (item) => this.data.thanks.map((thanks) => thanks.ownerId).indexOf(item.id) <= -1
-      );
+      this.owners = this.owners.filter((item) => this.data.thanks.includes(item));
```
Comment (xkureck) [inline apps/publications/src/app/dialogs/add-thanks-dialog/add-thanks-dialog.component.ts:45]: I dont think this filtering works, even if I insert something into the allowed_owners_for_thanks, there is no owner I can add
also I wanted to be the default value - when the field is empty - all owners can be thanked

File: apps/publications/src/app/pages/my-publications-page/my-publications-page.component.ts [hunk 4]
```diff
@@ -12,26 +12,28 @@ import { TABLE_PUBLICATION_AUTHOR_DETAIL_PUBLICATIONS } from '@perun-web-apps/co
 import { ActivatedRoute } from '@angular/router';
 import { MatDialog } from '@angular/material/dialog';
 
+interface Filter {
```
Comment (xkureck) [inline apps/publications/src/app/pages/my-publications-page/my-publications-page.component.ts:15]: dont you want to move this interface to its single file, so we don't have to duplicate?

**Review by xkureck — CHANGES_REQUESTED** [review]
when I go to authors or categories some error in console appears
image

File: apps/publications/src/app/pages/my-publications-page/my-publications-page.component.ts (hunk 4 above)
Comment (bodnara) [reply apps/publications/src/app/pages/my-publications-page/my-publications-page.component.ts]: removed duplicates

File: apps/publications/src/app/dialogs/add-thanks-dialog/add-thanks-dialog.component.ts (hunk 3 above)
Comment (bodnara) [reply apps/publications/src/app/dialogs/add-thanks-dialog/add-thanks-dialog.component.ts]: fixed filtering

File: apps/publications/src/app/dialogs/add-thanks-dialog/add-thanks-dialog.component.ts (hunk 2 above)
Comment (bodnara) [reply apps/publications/src/app/dialogs/add-thanks-dialog/add-thanks-dialog.component.ts:41]: fixed filtering

File: apps/publications/src/app/dialogs/add-authors-dialog/add-authors-dialog.component.ts (hunk 1 above)
Comment (bodnara) [reply apps/publications/src/app/dialogs/add-authors-dialog/add-authors-dialog.component.ts]: fixed filtering

**Comment by bodnara** [comment]
> when I go to authors or categories some error in console appears

This is not part of this issue nor it is connected to it.

**Review by xkureck — APPROVED** [review]

**Pull request merged**